std/crypto/tls/conn.go

```diff
@@ -1390,8 +1395,12 @@ func (c *Conn) handleKeyUpdate(keyUpdate *keyUpdateMsg) error {
 
 // Append .
 func (c *Conn) Append(b []byte) (int, error) {
-	c.closeMux.Lock()
-	defer c.closeMux.Unlock()
+	if !c.hasLock {
```

Comment:
This is another race condition, reading it outside the lock, then setting it inside the lock. probably should read it again inside the lock; but what would that mean if it has a value true after reading it as having a value of false?  having A recursive lock is probably the correct approach if this was c/c++ code.